exm/contract.js

```diff
@@ -1,9 +1,41 @@
-export default function handle(state, action) {
-  const { functionRole, title, url } = action.input;
+export function handle(state, action) {
+
+  /**
+   *****POST FORMAT*****
+     const post = {
+      postID: '',
+      author: '',
+      title: '',
+      url: '',
+      description: '',
+      dateAndTime: '',
+      comments: [
+        {
+          author: '',
+          text: '',
+          replies: [
+            {
+              author: '',
+              text: '',
+            }
+          ]
+        }
+      ]
+
+    }
+   */
+
+
+  const { functionRole, postID, author, title, url, description, timeCreated, upvotes, comments, replies } = action.input;
 
   if (functionRole === 'createPost') {
```

Comment:
use an enum. Actions shouldn't be string, they should be enums. You can do this in javascript or typescript

javascript:
```
const actions = { 
  CREATE_POST: 'createPost'
}
actions.CREATE_POST
```

typescript:
```
enum Actions {
CREATE_POST = "CREATE_POST"
}
```